internal/pkg/service/common/task/node.go

```diff
@@ -168,15 +164,15 @@ func (n *Node) StartTask(ctx context.Context, cfg Config) (t Task, err error) {
 }
 
 // RunTaskOrErr in foreground, the task run at most once, it is provided by local lock and etcd transaction.
-func (n *Node) RunTaskOrErr(cfg Config) error {
-	_, err := n.RunTask(cfg)
+func (n *Node) RunTaskOrErr(ctx context.Context, cfg Config) error {
+	_, err := n.RunTask(ctx, cfg)
 	return err
 }
 
 // RunTask in foreground, the task run at most once, it is provided by local lock and etcd transaction.
-func (n *Node) RunTask(cfg Config) (t Task, err error) {
+func (n *Node) RunTask(ctx context.Context, cfg Config) (t Task, err error) {
 	// Prepare task, acquire lock, handle error during prepare phase
-	task, fn, err := n.prepareTask(n.tasksCtx, cfg)
+	task, fn, err := n.prepareTask(ctx, cfg)
 	if err != nil {
 		return Task{}, err
```

Comment:
The prepare task phase is now using caller context, so it is now correctly reported by the telemetry.

Reply:
Task run phase has its own background context, see next commit.